src/hooks/useCalculatedColumns.ts

```diff
@@ -0,0 +1,252 @@
+import { useMemo } from 'react';
```

Comment:
Same as previous `useViewportHook.` I moved the last `useMemo` to the new `useViewport` hook as it needed rows